src/provider/client/index.ts

```diff
@@ -205,37 +207,62 @@ export class QualysAPIClient {
   }
 
   public async verifyAuthentication(): Promise<void> {
-    const endpoint = '/api/2.0/fo/activity_log/';
+    const { getPath, method } = ENDPOINTS.verifyAuthentication;
+    const path = getPath();
 
     try {
       await this.executeAuthenticatedAPIRequest(
-        this.qualysUrl(endpoint, {
+        this.qualysUrl(path, {
           action: 'list',
           username: this.config.qualysUsername,
           truncation_limit: 1,
         }),
         {
-          method: 'GET',
+          method,
         },
       );
     } catch (err) {
       throw new IntegrationProviderAuthenticationError({
         cause: err,
-        endpoint,
+        endpoint: path,
         status: err.status,
         statusText: err.statusText,
       });
     }
   }
 
+  public async verifyApiExistance(): Promise<void> {
+    for (const name in ENDPOINTS) {
+      try {
+        const { getPath, method } = ENDPOINTS[name];
+        const path = getPath(); // uses 'undefined' for all path params
+        await this.executeAPIRequest(this.qualysUrl(path), {
+          timeout: 0,
+          headers: {
+            method,
+          },
+        });
+      } catch (err) {
+        const unauthorized = 401; // expecting unauthorized as no auth is sent
+        const ok = 200; // some endpoints put unauthorized response in body
+        if (![ok, unauthorized].includes(Number(err.status))) {
+          throw new IntegrationValidationError(
```

Comment:
Aside from `IntegrationValidationError`, it would be helpful for us as operators to be able to inspect the error thrown here. Unfortunately, `IntegrationValidationError` isn't great for passing error details to users/operators.

I notice that this client does not have a `logger` in its context; it would be nice to add one to the constructor and then use it here to expose this information.

```suggestion
          logger.warn({ err }, 'Error validating a Qualys API endpoint.');

          throw new IntegrationValidationError(
```

Reply:
There seems to be some discrepancy between you and Adam on if IntegrationValidationErrors log to CloudWatch or not. I'll investigate to be sure before committing this.